src/cli/msig/index.ts

```diff
@@ -164,6 +168,177 @@ export default async function msig(args: string[]) {
+
+            // 2. Prepare SETABI
+            const abi = JSON.parse(abiFile);
+            const abiDef = ABI.from(abi);
+            const abiSerializedHex = Serializer.encode({ object: abiDef }).hexString;
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            const actions: ActionData[] = [];
+
+            // Prepare SETCODE action
+            const setCodeAction: ActionData = {
+                account: 'tonomy',
+                name: 'setcode',
+                authorization: [
+                    {
+                        actor: contractName.toString(),
+                        permission: 'active',
+                    },
+                ],
+                data: {
+                    account: contractName.toString(),
+                    vmtype: 0,
+                    vmversion: 0,
+                    code: wasm,
+                },
+            };
+
+            actions.push(setCodeAction);
+
+            // Prepare SETABI action
+            const setAbiAction: ActionData = {
+                account: 'tonomy',
+                name: 'setabi',
+                authorization: [
+                    {
+                        actor: contractName.toString(),
+                        permission: 'active',
+                    },
+                ],
+                data: {
+                    account: contractName.toString(),
+                    abi: abiSerializedHex,
+                },
+            };
+
+            actions.push(setAbiAction);
+
+            const proposalHash = await createProposal(
+                proposer,
+                proposalName,
+                actions,
+                privateKey,
+                newGovernanceAccounts
+            );
+
+            if (test) await executeProposal(proposer, proposalName, proposalHash);
+        } else if (proposalType === 'eosio.code-permission') {
+            let count = 1;
+
+            console.log('addCodePermissionTo', addCodePermissionTo, addCodePermissionTo.length);
+
+            for (const account of addCodePermissionTo) {
+                count = count + 1;
+
+                console.log('accountName', account);
+                const accountInfo = await getAccountInfo(Name.from(account));
+
+                const ownerPermission = accountInfo.getPermission('owner');
+                const activePermission = accountInfo.getPermission('active');
+
+                console.log('permissionss', ownerPermission, activePermission);
+
+                // const ownerAuthority = Authority.fromAccount({
+                //     actor: account,
+                //     permission: ownerPermission.perm_name.toString(),
+                // });
+
+                // const activeAuthority = Authority.fromAccount({
+                //     actor: account,
+                //     permission: activePermission.perm_name.toString(),
+                // });
+                const ownerAuthority = new Authority(
```

Comment:
lets make a new constructor overload `Authory.fromPermission(permission: AccountPermission)` that will do this logic here to create the Authority object